13.go

```diff
@@ -989,10 +1021,10 @@ func (hs *clientHandshakeState) doTLS13Handshake() error {
 
 	// 0-RTT is not supported yet, so use an empty PSK.
 	hs.keySchedule.setSecret(nil)
-	ecdheSecret := c.deriveDHESecret(serverHello.keyShare, hs.privateKey)
-	if ecdheSecret == nil {
+	ecdheSecret, err := c.keyAgreementClient(serverHello.keyShare, hs.privateKey)
```

Comment:
Nit: change variable name (to remove dhe)?

Reply:
let's make it ``sharedSecret`` as that's what it is